### PR TITLE
Hooks aggregation

### PR DESCRIPTION
**Pull request by dmzmk**
Adds ability to aggregate hook calls on demand
Small demo:
Screenshot 2022-06-15 at 21 30 10

### CODE REVIEW

File: packages/wasm/src/lib.rs [hunk 1]
```diff
@@ -10,10 +12,59 @@ extern "C" {
     fn log(_: &str);
 }
 
+/// Aggragation state
```
Comment (tbaums) [inline packages/wasm/src/lib.rs:15]: *Aggregation

**Review by tbaums — APPROVED** [review]
From an English language perspective, LGTM. If there's something in particular  you'd like me to review or are concerned about, pls let me know.

File: packages/core/src/index.ts [hunk 2]
```diff
@@ -41,20 +41,28 @@ export class Instance implements Instance {
...
+            ])
+          }
         }
       }
-    }
+    })
   }
 }
 
+function with_aggregate(log: Index.List, cb: () => void) {
```
Comment (artps) [inline packages/core/src/index.ts:60]:
```suggestion
function withAggregate(log: Index.List, cb: () => void) {
```

File: packages/core/src/index.ts [hunk 3]
```diff
@@ -41,20 +41,28 @@ export class Instance implements Instance {
...
   commit(entities: Array<any>) {
-    for (const entity of entities) {
-      if (entity.$id) {
-        for (const [attr, value] of Object.entries(entity)) {
-          this.log.append([
-            entity.$id,
-            attr,
-            value
-          ])
+    with_aggregate(this.log, () => {
```
Comment (artps) [inline packages/core/src/index.ts:44]:
```suggestion
    withAggregate(this.log, () => {
```

File: packages/wasm/src/lib.rs (hunk 1 above)
Comment (artps) [inline packages/wasm/src/lib.rs:15]:
```suggestion
/// Aggregation state
```

File: packages/core/src/index.ts [hunk 4]
```diff
@@ -41,20 +41,28 @@ export class Instance implements Instance {
...
+          }
         }
       }
-    }
+    })
   }
 }
 
+function with_aggregate(log: Index.List, cb: () => void) {
+  log.aggregate_hooks(true);
```
Comment (artps) [inline packages/core/src/index.ts:61]:
```suggestion
  log.aggregateOps(true);
```

By change, two suggestions:

1. I believe, here you don't really aggregating the hooks, but ops.
2. May we start doing something like `#[wasm_bindgen(js_name = aggregateOps)]`?

**Pull request merged**